Sources/AppcuesKit/Presentation/Debugger/ScreenCapturing/UIDebugger+Capture.swift

```diff
@@ -0,0 +1,171 @@
+//
+//  UIDebugger+Capture.swift
+//  AppcuesKit
+//
+//  Created by James Ellis on 2/16/23.
+//  Copyright © 2023 Appcues. All rights reserved.
+//
+
+import Foundation
+
+internal enum ScreenCaptureError: Error {
+    case customerAPINotFound
+    case noImageData
+    case noSelf
+    case noImageURL
+    case failedCaptureEncoding
+}
+
+@available(iOS 13.0, *)
+extension UIDebugger {
+
+    func saveScreenCapture(networking: Networking,
+                           screen: Capture,
+                           authorization: Authorization,
+                           completion: @escaping (Result<Void, Error>) -> Void) {
+
+        // this is a 4-step chain of nested async completion blocks
+        // each step will execute and call on to the next, if successful, or bubble up error if not
+
+        // start the chain by looking up the customer API path
+        getCustomerAPIHost(networking: networking, screen: screen, authorization: authorization, completion: completion)
+    }
+
+    // 1. service discovery -> customer API
+    private func getCustomerAPIHost(networking: Networking,
+                                    screen: Capture,
+                                    authorization: Authorization,
+                                    completion: @escaping (Result<Void, Error>) -> Void) {
+        networking.get(from: SettingsEndpoint.settings,
```

Comment:
yeah pondered the same and just did the simple thing to start - but yes we could totally keep in Storage or something like that. I figured this one was heavily cached at network layer anyway and may not matter much, or may be doing URLSession layer caching already, but I did not spend time yet to figure out the details.

Since it is a manually initiated screen capture that we presume somewhat rare - i agree we can cycle back to it.